test/uri/uri_recompose_test.cc

```diff
@@ -35,3 +35,88 @@ TEST(URI_recompose, empty_fragment) {
   const sourcemeta::jsontoolkit::URI uri{"#"};
   EXPECT_EQ(uri.recompose(), "#");
 }
+
+/*
+ * RFC 3986 normalization tests
+ */
+
+// Inspired from https://cr.openjdk.org/~dfuchs/writeups/updating-uri/
+
+TEST(URI_normalize, rfc3986_1) {
```

Comment:
Actually, can we call these `URI_recompose`, as the test is really exercising normalisation _through_ recomposing?

Reply:
I am not sure tho! haha!

For me what we are testing is **normalisation** even if we test it through the recompose API. By the way I think that it is a good way to differentiate theses tests from what we have before.

WDYT?

Reply:
This is a preference tho! If it hurts you Im a okay to rename it :)

Reply:
So then let's have a `uri_normalize_test.cc` file with these? The current convention is that the test category matches the module + file, so a new file could work well?

Reply:
sold! I started with this apporach btw haha :)

Reply:
Haha awesome